ads/custom.md

```diff
@@ -0,0 +1,172 @@
+```
+
+### Two ads with different slots
+
+```html
+<amp-ad width=300 height=250
+    type="custom"
+    data-slot="1"
+    data-url="https://mysite/my-ad-server"
+>
+    <template type="amp-mustache" id="amp-template-id">
+      <a href="{{href}}">
+        <amp-img layout='fixed' height="300" width="250" src="{{src}}" data-info="{{info}}"></amp-img>
+      </a>
+    </template>
+</amp-ad>
+<amp-ad width=400 height=300
+    type="custom"
+    data-slot="2"
+    data-url="https://mysite/my-ad-server"
+>
+    <template type="amp-mustache" id="amp-template-id">
+      <a href="{{href}}">
+        <amp-img layout='fixed' height="400" width="300" src="{{src}}" data-info="{{info}}"></amp-img>
+      </a>
+    </template>
+</amp-ad>
+<!-- The ad server will be called with the URL https://mysite/my-ad-server?s=1,2 -->
+```
+
+### Ads from different ad servers
+```html
+<amp-ad width=300 height=250
+    type="custom"
+    data-slot="slot-name-a"
+    data-url="https://mysite/my-ad-server"
+>
+    <template type="amp-mustache" id="amp-template-id">
+      <a href="{{href}}">
+        <amp-img layout='fixed' height="300" width="250" src="{{src}}" data-info="{{info}}"></amp-img>
+      </a>
+    </template>
+</amp-ad>
+<amp-ad width=400 height=300
+    type="custom"
+    data-slot="slot-name-b"
+    data-url="https://mysite/my-ad-server"
+>
+    <template type="amp-mustache" id="amp-template-id">
+      <a href="{{href}}">
+        <amp-img layout='fixed' height="400" width="300" src="{{src}}" data-info="{{info}}"></amp-img>
+      </a>
+    </template>
+</amp-ad>
+<amp-ad width=300 height=250
+    type="custom"
+    data-slot="123"
+    data-url="https://my-other-site/my-other-ad-server"
+>
+    <template type="amp-mustache" id="amp-template-id">
+      <a href="{{href}}">
+        <amp-img layout='fixed' height="300" width="250" src="{{src}}" data-info="{{info}}"></amp-img>
+      </a>
+    </template>
+</amp-ad>
+<!-- Two ad server calls will be made: -->
+<!-- The first:  https://mysite/my-ad-server?s=slot-name-a,slot-name-b -->
+<!-- The second: https://my-other-site/my-other-ad-server?s=123 -->
+```
+
+## Supported parameters
+
+### data-url (mandatory)
+
+This must be starting with `https://`, and it must be the address of an ad
+server returning json in the format defined below.
+
+### data-slot (optional)
+
+On the assumption that most pages have multiple ad slots, this is passed to the
+ad server to tell it which slot is being fetched. This can be any alphanumeric string.
+
+## Ad server
+
+The ad server will be called once for each value of `data-url` on the page: for the vast 
+majority of applications, all your ads will be from a single server so it will be
+called only once.
+
+A parameter like `?ampslots=1,2` will be appended to the URL specified by `data-url` in order
+to specify the slots being fetched. See the examples above for details.
+
+The ad server should return a json object containing a record for each slot in the request.
```

Comment:
could you mention the JSON object is `keyed by the provided slot ID`.

and for single ad request with no slot ID provided, the response is simply the ad data.